judge/views/submission.py

```diff
@@ -179,8 +179,12 @@ def get(self, request, *args, **kwargs):
 @require_POST
 def abort_submission(request, submission):
     submission = get_object_or_404(Submission, id=int(submission))
-    if (not request.user.is_authenticated or (submission.was_rejudged or (request.profile != submission.user)) and
-            not request.user.has_perm('judge.abort_any_submission')):
+    if not request.user.is_authenticated:
+        raise PermissionDenied()
+    if (
+        not request.user.has_perm('judge.abort_any_submission') and
+        (submission.rejudged_date is not None or request.profile != submission.user)
+    ):
```

Comment:
We typically use this style.

```suggestion
    if (request.user.has_perm('judge.abort_any_submission') and
        (submission.rejudged_date is not None or request.profile != submission.user)):
```